deepchecks/recommender/checks/user_session_length_drift.py

```diff
@@ -0,0 +1,139 @@
+from deepchecks.recommender import Context
+from deepchecks.tabular import TrainTestCheck
+from deepchecks.utils.distribution.drift import calc_drift_and_plot
+from deepchecks.utils.validation import is_sequence_not_str
+
+__all__ = ['UserSessionDrift']
+
+class UserSessionDrift(TrainTestCheck):
+    """
+    
+    Check for user session length drift in train and test sets. This check compares the distribution of user session lengths between the train and test sets to identify potential drift or discrepancies. It helps assess whether the two sets exhibit similar patterns in terms of user session lengths or if there are significant differences that could introduce drift in the recommender system.
+    
+    """
+    def __init__(
+            self,
+            margin_quantile_filter: float = 0.025,
+            max_num_categories_for_drift: int = None,
+            min_category_size_ratio: float = 0.01,
+            max_num_categories_for_display: int = 10,
+            show_categories_by: str = 'largest_difference',
+            numerical_drift_method: str = 'KS',
+            categorical_drift_method: str = 'cramers_v',
+            balance_classes: bool = False,
+            ignore_na: bool = True,
+            aggregation_method: t.Optional[str] = 'max',
+            min_samples: t.Optional[int] = 10,
+            n_samples: int = 100_000,
+            random_state: int = 42,
+            **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.margin_quantile_filter = margin_quantile_filter
+        self.max_num_categories_for_drift = max_num_categories_for_drift
+        self.min_category_size_ratio = min_category_size_ratio
+        self.max_num_categories_for_display = max_num_categories_for_display
+        self.show_categories_by = show_categories_by
+        self.numerical_drift_method = numerical_drift_method
+        self.categorical_drift_method = categorical_drift_method
+        self.balance_classes = balance_classes
+        self.ignore_na = ignore_na
+        self.aggregation_method = aggregation_method
+        self.min_samples = min_samples
+        self.n_samples = n_samples
+        self.random_state = random_state
+        if self.aggregation_method not in ('weighted', 'mean', 'none', None, 'max'):
+            raise DeepchecksValueError('aggregation_method must be one of "weighted", "mean", "max", None')
+
+    def run_logic(self, context: Context) -> CheckResult:
+        """Calculate drift for all columns.
+
+        Returns
+        -------
+        CheckResult
+            value: drift score.
+            display: label distribution graph, comparing the train and test distributions.
+        """
+
+        train_dataset = context.train
+        test_dataset = context.test
+
+        user_col = train_dataset.user_index_name
+        assert user_col == test_dataset.user_index_name
+
+        train_user_session_length = train_dataset.data[user_col].value_counts().tolist()
```

Comment:
That's not session lenght, that's the amount of interactions each user has no?

Reply:
fixed https://github.com/deepchecks/deepchecks/pull/2610/commits/f9e1f6c21cdfdb3cc113bcd1744f3c7b3629b1c9